generators/base-application/application-config-all.d.ts

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable @typescript-eslint/consistent-type-imports */
+import type { Simplify } from 'type-fest';
+import type { ExportStoragePropertiesFromCommand } from '../../lib/command/types.js';
+import type { YO_RC_CONFIG_KEY } from '../../lib/utils/yo-rc.ts';
+import type { Config } from '../base/types.js';
+
+type BaseApplicationConfig = {
+  entities?: string[];
+  backendType?: string;
+  projectDescription?: string;
+};
+
+export type ApplicationConfiguration = Simplify<
```

Comment:
I do not see how moving these classes to this generator helps in any way, still, let's make it